src/tasks/block/sim.rs

```diff
@@ -158,8 +156,8 @@ impl Simulator {
             block_number = built_block.block_number(),
             "block simulation completed",
         );
-        counter!("signet.builder.built_blocks").increment(1);
-        histogram!("signet.builder.built_blocks.tx_count").record(built_block.tx_count() as u32);
+        metrics::built_blocks().increment(1);
+        metrics::built_blocks_tx_count().record(built_block.tx_count() as u32);
```

Comment:
these lines should both be collapsed into `metrcis::record_bulit_block(&block)` and the logic to increment, etc should be in the `metrics` file